Ednpoints.txt

```diff
@@ -0,0 +1,132 @@
+The same routes for users, chefs and delivery
+POST: /auth/register - registration
+POST: /auth/login - enter to account
+GET: /auth/logout - exit from account
+POST: /auth/forgot-password - request to change password
+POST: /auth/reset-password - password reset
+
+/************/
+
+USER ENDPOINTS:
+
+GET: /users/:userId - get user's info
+PATCH: /users/:userId - update user's info
+DELETE: /users/:userId - delete user's account
+
+GET: /dishes - get a list of dishes
+GET: /dishes/:dishId - get dish information
+GET: /dishes/popular - get popular dishes
+GET: /dishes/random - get some random dish
+
+GET: /chefs - get all chefs
+
+GET: /users/:userId/favourites?type=chef(dish) - get user's favourite dishes or chefs
+	or /users/:userId/favourite-dishes and /users/:userId/favourite-chefs
+POST, DELETE:
+/users/:userId/favourites?type=dishes - add/delete the dish to favourites (passing dishId)
+/users//:userId/favourites?type=chefs - add/delete the chef to favourites (passing chefId)
+
+**user's cart will be stored on the server instead of in local storage
+GET: /users/:userId/cart - get the cart info
+POST: /users/:userId/cart - add the dish to the cart (passing dishId)
+PATCH: /users/:userId/cart - update information about the dish in the cart (passing dishId)
+DELETE: /users/:userId/cart  - delete the dish from the cart (passing dishId)
+
+GET: /notifications/:userId - get notifications of this user
+
+POST: /reviews - create a new comment to this dish (pass dishId and userId)
+PATCH: /reviews/:reviewId - update user's comment to this dish
+DELETE: /reviews/:reviewId - delete user's comment to this dish
+
+GET: /orders/:userId - get user's orders list
+GET: /orders/:orderId - get a particular order
+POST: /orders - add a new order to user's orders list
+
+/************/
+
+CHEF ENDPOINTS:
+
+GET: /chefs/:chefId - get chef's info
+PATCH: /chefs/:chefId - update chef's info
+DELETE: /chefs/:chefId - delete chef's account
+
+GET: /dishes/:chefId - get all dishes of this chef
```

Comment:
it is better to use something like `/dishes/own`
current approach will create collision with `/dishes/:dishId` because of the same endpoint structure

according to express, you will hit an endpoint that is placed first by default
so, it you have `/dishes/:chefId` first, but initiate a request for `/dishes/:dishId`, you will hit **chef** endpoint and get `NOT_FOUND` error, because we don't have a chef with needed ID

Reply:
Yeah, we messed up here( Thanks